index.js

```diff
@@ -509,7 +509,7 @@ function maybeUnzip (payload, serialize) {
   // handle case where serialize doesn't return a string or Buffer
   if (!Buffer.isBuffer(buf)) return result
   if (isCompressed(buf) === 0) return result
-  return intoStream(result)
+  return Readable.from(intoAsyncIterator(result))
```

Comment:
You called `Readable.from(intoAsyncIterator(result))` 3 times. 
So I'm wondering if it makes sense to export `intoReadable(result)` (or you decide a better name) instead of `intoAsyncIterator`

But you might need to revise the unittests.

Reply:
v8 is good at optimizing the function that always receives the same input. So, instead of two functions to receive different input. One is enough.